src/common/util.c

```diff
@@ -45,6 +47,7 @@
 #endif
 
 #define PATH_SEPARATOR '/'
+#define OSCAP_PCRE_EXEC_RECURSION_LIMIT 1000
```

Comment:
The question is: How big should the limit be?
On my machine, the segfault happened at ~ 17600.

Reply:
I wonder if this number is a derivative of the the string or the pattern.

Reply:
In order to determine where we should draw the line I would start with answering the question of how many recursions is takes to match the content which actually matches (from the set of strings/patterns we have to deal with).

Reply:
It probably is, but I don't know how to compute it.

There can be arbitrary string and arbitrary valid regular expression.

Reply:
Default is: 10 000 000. ModSecurity, for example, has it at 1500. I guess it'd be hard to properly estimate the value.

Maybe we could make it an option for the scanner instead? Then users could adjust it in case it is too small for their needs, without rebuilding the scanner. Just a wild idea, it might be not easy to implement.

Reply:
Great idea, we can do that, or we can add possibility to modify this by environment variable. 

However, we need to set a sensible default, because I assume most of the users will use the default.

Reply:
Oh, I think that it would be your favourite :) Number guessing :)). I'd vote for 5000. Rationale:  ~⅓ of your machine's limit on a set of content we have.

Reply:
I'm fine with 5000